qiskit/transpiler/passes/basis/decompose.py

```diff
@@ -107,6 +116,9 @@ def _should_decompose(self, node) -> bool:
         if self.gates_to_decompose is None:  # check if no gates given
             return True
 
+        if node.op.name == "state_preparation":  # not sure about other gate also cant pass
+            return True
+
```

Comment:
The pass shouldn't have this sort of special handling - it overrides the configuration input to the pass.

Reply:
may be add a test for ``_should_decompose``

Reply:
because I don't see any reason to not let it decompose ``gates_to_decompose`` repeatly.

Reply:
sry, I don't understand what "overrides the configuration input to the pass." mean.